libraries/chain/include/eosio/chain/authority_checker.hpp

```diff
@@ -225,19 +225,18 @@ namespace detail {
                      bool r = false;
                      typename permission_cache_type::iterator itr = cached_permissions.end();
 
-                     bool propagate_error = false;
+                     std::invoke_result_t<decltype(checker.permission_to_authority), const permission_level> auth = nullptr;
```

Comment:
This is pretty weird; it comes about because `permission_to_authority` could return either a `shared_authority` or an `authority` (well, now either a `shared_authority*` or `authority*`). I want the scope to be outside the `try` to make handling the null check more straightforward so I couldn't use `auto` in this particular pattern.